shortfin/python/shortfin_apps/llm/components/kvcache/trie_attention_cache.py

```diff
@@ -455,6 +508,17 @@ def free_cache_pages(self):
             self.page_pool.free_pages(pages_to_free)
 
         self.page_pool.free_pages(self._allocated_pages)
+        self._allocated_pages = []
+
+    def free_allocated_pages(self, page_ids: List[int]):
+        pages = []
+        for id in page_ids:
```

Comment:
Double nest loops feel concerning - is there a better way we can do this (feels like it would be computationally expensive for large caches)